spec/builtin-process.test.sh

```diff
@@ -142,6 +143,135 @@ stat -c '%a' $TMP/umask-one $TMP/umask-two
 ## END
 ## stderr-json: ""
 
+#### umask -p
+umask -p | grep 'umask [0-9][0-9][0-9][0-9]'
+## status: 0
+## N-I dash/mksh/zsh status: 1
+
+#### umask -S
+# current mask as symbolic
+umask -S | grep 'u=[rwx]*,g=[rwx]*,o=[rwx]*' 
+## status: 0
+
+#### umask symbolic parsing
+
+umask 0000
+umask u-rw
+echo status0=$?
+umask | tail -c 4
+
+umask 0700
+umask u=r
+echo status1=$?
+umask | tail -c 4
+
+umask 0000
+umask u=r,g=w,o=x
+echo status2=$?
+umask | tail -c 4
+
+umask 0777
+umask u+r,g+w,o+x
+echo status3=$?
+umask | tail -c 4
+
+umask 0000
+umask u-r,g-w,o-x
+echo status4=$?
+umask | tail -c 4
+
+umask 0137
+umask u=,g+,o-
+echo status5=$?
+umask | tail -c 4
+
+## status: 0
+## STDOUT:
+status0=0
+600
+status1=0
+300
+status2=0
+356
+status3=0
+356
+status4=0
+421
+status5=0
+737
+## END
+
+#### umask symbolic parsing with spaces
+umask 0111
+# spaces are an error in bash
+# dash & mksh only interpret the first one
+umask u=, g+, o-
+echo status=$?
+umask | tail -c 4
+## status: 0
+## STDOUT:
+status=2
+111
+## END
+## OK bash/zsh STDOUT:
+status=1
+111
+## END
+## BUG dash/mksh STDOUT:
+status=0
+711
+## END
+
+#### umask bad symbolic input
+umask b=rwx
+## status: 1
+## OK dash status: 2
+
+#### umask octal number out of range
+umask 0022
+umask 1234567
+# osh currently treats 0o1234567 as 0o0567
```

Comment:
Can we handle this correctly in OSH?

If there is something TODO, then we just let OSH fail with oils_allowed_failures.

We don't mark OSH as a BUG -- that is only for non-OSH shells!